### DISCUSSION TITLE
pyinfra community modules

### DISCUSSION THREAD

**Discussion by apecnascimento** [pyinfra-dev/pyinfra#1099] [Q&A]
In Ansible we have the community.general collection to group community modules, making the cycle of creating new modules occur in parallel with the Ansible Core functionalities. Is there any equivalent initiative in pyinfra, so that the community can create connectors, Facts and operators in addition to those in the pyinfra core?

**Comment by xvello** [comment]
Heya @apecnascimento, it seems like pyinfra has not yet reached the size where a split between core & community makes sense, the author encourages [contributing new operations and facts to the core](https://docs.pyinfra.com/en/3.x/contributing.html) for now. Looks like there's [several such PRs](https://github.com/pyinfra-dev/pyinfra/pulls?q=is%3Apr+add).

**Comment by Fizzadar** [comment]
Very much open to thoughts here, would happily start a community repo if preferred. But also having everything builtin is actually quite a good experience I think.

Thinking back over the years this hasn’t been much of an issue, mainly because the operation testing tends to be strong enough to prevent regressions (operations very rarely change).

**Reply by apecnascimento** [reply]
After create some facts and operations in pyinfra I agree with @Fizzadar. I think what pyinfra really need is increase a number of operations and facts and I'll contribute with that.

**Comment by erlangparasu** [comment]
how about deploys? https://docs.pyinfra.com/en/3.x/api/deploys.html

**Reply by Fizzadar** [reply]
For deploys I think Python packages is the way to go - no point reinventing the wheel. An example: https://github.com/oxygem/pyinfra-tinc

**Reply by apecnascimento** [reply]
My only difficulty with pyinfra deploys is figuring out if a deploy already exists for what need. With that in mind, I think it would be great if the docs had a deploys index page with links pointing to the pypi or github page that describes the respective deploy.
